modules/xmpp/ChatRoom.js

```diff
@@ -893,15 +896,37 @@ export default class ChatRoom extends Listenable {
      * @param handler
      */
     addPresenceListener(name, handler) {
-        this.presHandlers[name] = handler;
+        if (typeof handler !== 'function') {
+            throw new Error('"handler" is not a function');
+        }
+        let tagHandlers = this.presHandlers[name];
+
+        if (!tagHandlers) {
+            this.presHandlers[name] = tagHandlers = [];
+        }
+        if (tagHandlers.indexOf(handler) === -1) {
+            tagHandlers.push(handler);
+        } else {
+            logger.warn(
+                `Trying to add the same handler more than once for: ${name}`);
+        }
     }
 
     /**
      *
      * @param name
+     * @param handler
      */
-    removePresenceListener(name) {
-        delete this.presHandlers[name];
+    removePresenceListener(name, handler) {
+        const tagHandlers = this.presHandlers[name];
+        const handlerIdx = tagHandlers ? tagHandlers.indexOf(handler) : -1;
+
+        // eslint-disable-next-line no-negated-condition
```

Comment:
For me -1 means does not exists in the array. Having to compare that to >= 0 will make me think and trying to decode what does it mean. I would prefer to keep it this way.